### PR TITLE
[SecurityBundle] Removing test insulations for a huge perf win

### PR DESCRIPTION
**Pull request by nicolas-grekas**
| Q | A |
| --- | --- |
| Bug fix? | no |
| New feature? | no |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets | - |
| License | MIT |
| Doc PR | - |

### CODE REVIEW

**Comment by nicolas-grekas** [comment]
With this PR, HHVM tests last 5 minutes instead of 10 on Travis!

**Comment by fabpot** [comment]
:-1: Insulation ensures that there is no side-effects between tests.

**Comment by nicolas-grekas** [comment]
If you look at the patch, insulation still exists: not at the process level, but at the kernel level, which is enough to ensure no side-effects between tests.

File: src/Symfony/Bundle/SecurityBundle/Tests/Functional/app/AppKernel.php [hunk 1]
```diff
@@ -63,6 +63,8 @@ public function __construct($testCase, $rootConfig, $environment, $debug)
         $this->rootConfig = $rootConfig;
 
         parent::__construct($environment, $debug);
+
+        $this->name = $this->getName().md5($this->rootConfig);
```
Comment (nicolas-grekas) [inline src/Symfony/Bundle/SecurityBundle/Tests/Functional/app/AppKernel.php]: this is the only insulation that is required to ensure tests have no overlap

**Comment by nicolas-grekas** [comment]
ping @stof @fabpot there is real benefit into this

**Comment by fabpot** [comment]
Just removed my veto.

File: src/Symfony/Bundle/SecurityBundle/Tests/Functional/FormLoginTest.php [hunk 2]
```diff
@@ -84,17 +81,13 @@ public function getConfigs()
         );
     }
 
-    protected function setUp()
+    public static function setUpBeforeClass()
     {
-        parent::setUp();
-
-        $this->deleteTmpDir('StandardFormLogin');
+        parent::deleteTmpDir('CsrfFormLogin');
```
Comment (Tobion) [inline src/Symfony/Bundle/SecurityBundle/Tests/Functional/FormLoginTest.php]: looks like the wrong one

File: src/Symfony/Bundle/SecurityBundle/Tests/Functional/FirewallEntryPointTest.php [hunk 3]
```diff
@@ -46,17 +44,13 @@ public function testItUsesTheConfiguredEntryPointFromTheExceptionListenerWithFor
         );
     }
 
-    protected function setUp()
+    public static function setUpBeforeClass()
     {
-        parent::setUp();
-
-        $this->deleteTmpDir('FirewallEntryPoint');
+        parent::deleteTmpDir('CsrfFormLogin');
```
Comment (Tobion) [inline src/Symfony/Bundle/SecurityBundle/Tests/Functional/FirewallEntryPointTest.php]: looks like the wrong one

File: src/Symfony/Bundle/SecurityBundle/Tests/Functional/LocalizedRoutesAsPathTest.php [hunk 4]
```diff
@@ -81,17 +77,13 @@ public function getLocales()
         return array(array('en'), array('de'));
     }
 
-    protected function setUp()
+    public static function setUpBeforeClass()
     {
-        parent::setUp();
-
-        $this->deleteTmpDir('StandardFormLogin');
+        parent::deleteTmpDir('CsrfFormLogin');
```
Comment (Tobion) [inline src/Symfony/Bundle/SecurityBundle/Tests/Functional/LocalizedRoutesAsPathTest.php]: looks like the wrong one

Comment (nicolas-grekas) [inline src/Symfony/Bundle/SecurityBundle/Tests/Functional/LocalizedRoutesAsPathTest.php]: good catch thanks, fixed

File: src/Symfony/Bundle/SecurityBundle/Tests/Functional/app/AppKernel.php (hunk 1 above)
Comment (Tobion) [inline src/Symfony/Bundle/SecurityBundle/Tests/Functional/app/AppKernel.php]: IMO it's cleaner to overwrite the `getName` method. The problem is that `name` is protected when it should actually be private which is a design flaw. There should not be two extension point for the same thing (public method & protected property).

Comment (nicolas-grekas) [inline src/Symfony/Bundle/SecurityBundle/Tests/Functional/app/AppKernel.php]: getName added (not really clean either :) )

File: src/Symfony/Bundle/SecurityBundle/Tests/Functional/CsrfFormLoginTest.php [hunk 5]
```diff
@@ -113,17 +109,13 @@ public function getConfigs()
         );
     }
 
-    protected function setUp()
+    public static function setUpBeforeClass()
     {
-        parent::setUp();
-
-        $this->deleteTmpDir('CsrfFormLogin');
+        parent::deleteTmpDir('CsrfFormLogin');
```
Comment (stof) [inline src/Symfony/Bundle/SecurityBundle/Tests/Functional/CsrfFormLoginTest.php]: why ``parent::`` rather than ``self::`` ?

Comment (nicolas-grekas) [inline src/Symfony/Bundle/SecurityBundle/Tests/Functional/CsrfFormLoginTest.php]: to hint the reader that the implementation is in the parent?

**Comment by Tobion** [comment]
LGTM :+1:

**Comment by fabpot** [comment]
Thank you @nicolas-grekas.

**Pull request merged**